lib/internal/modules/cjs/loader.js

```diff
@@ -483,6 +491,7 @@ Module._load = function(request, parent, isMain) {
   }
 
   if (experimentalModules && isMain) {
+    if (asyncESM === undefined) lazyLoadESM();
     asyncESM.loaderPromise.then((loader) => {
```

Comment:
I don't think this one requires everything that loads? Mind you, it might just end up getting loaded anyways.

Reply:
I did not specifically test for this case but if `esm` is used everything should be necessary (at least if more than one file is imported). This is just convenience in this case.